File: app/src/main/java/com/zxy/demo/http/ServiceInterface.java
package com.zxy.demo.http;

import com.zxy.demo.entity.SquarePostEntity;
import com.zxy.demo.entity.VersionUpdateEntity;
import com.zxy.frame.net.ApiBaseResponse;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Streaming;
import retrofit2.http.Url;

/**
 * Created by zxy on 2019/08/06.
 */
public interface ServiceInterface {

    String BASE_URL = "http://api.dev.ssbf.imjihua.com";
    String GET_TAB = "/api/get_tab";
    String SQUARE_POST = "/api/square_post";
    String VERSION_CHECK = "/api/version_check";

    @Streaming
    @GET
    Observable<ResponseBody> downloadFile(@Header("RANGE") String downParam, @Url String fileUrl);

    @POST(GET_TAB)
    Call<ResponseBody> get_tab();

    /**
     * 社区广场列表
     *
     * @param page
     * @return
     */
    @FormUrlEncoded
    @POST(SQUARE_POST)
    Observable<ApiBaseResponse<SquarePostEntity>> square_post(@Field("page") int page);

    /**
     * 版本更新
     *
     * @return
     */
    @POST(VERSION_CHECK)
    Observable<ApiBaseResponse<VersionUpdateEntity>> version_check();

}

File: app/src/main/java/com/zxy/demo/ServiceInterface.kt
package com.zxy.demo

import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.*
import java.io.File

/**
 * Created by Admin
 * on 2020/8/13
 */
interface ServiceInterface {

    companion object {
        var BASE_URL = "http://www.httpbin.org/"
    }

    @Streaming
    @GET
    fun downloadFile(
        @Header("RANGE") downParam: String,
        @Url fileUrl: String
    ): Call<ResponseBody>

    @GET
    fun getUrl(@Url url: String): Call<ResponseBody>

    @GET("/get")
    fun get(): Call<ResponseBody>

    @GET("/get")
    fun get(@QueryMap map: Map<String,String>): Call<ResponseBody>

    @GET("/get")
    fun get(@QueryName name:String): Call<ResponseBody>

    @POST("/post")
    fun post(): Call<ResponseBody>

    @POST("/post")
    @FormUrlEncoded
    fun post(@Field("name") name:String,@Field("content") content:String): Call<ResponseBody>

    @POST("/post")
    @Multipart
    fun post(@Part("va") va:RequestBody,@Part("name")name: File, @Part("name2")name2: File): Call<ResponseBody>

    @PATCH("/patch")
    @FormUrlEncoded
    fun patch(@Field("name") name:String,@Field("content") content:String): Call<ResponseBody>

    @DELETE("/delete")
    fun delete(): Call<ResponseBody>
}
